### ISSUE TITLE
help for installing OpenZwave

### ISSUE THREAD

**Issue by NickBue** [OpenZWave/node-red-contrib-openzwave#61]
Hi,

I'm not able to install the OpenZwave node. Not via the 'pallet' and no via CL.

Can someone please get me on the way. 
Thanks

Nick

> pi@raspberrypi:~/.node-red $ npm install node-red-contrib-openzwave
> 
> > openzwave-shared@1.4.5 preinstall /home/pi/.node-red/node_modules/.staging/openzwave-shared-32881044
> > node lib/install-ozw.js
> 
> 
> > openzwave-shared@1.4.5 install /home/pi/.node-red/node_modules/openzwave-shared
> > node-gyp rebuild
> 
> make: Map '/home/pi/.node-red/node_modules/openzwave-shared/build' wordt binnengegaan
>   CXX(target) Release/obj.target/openzwave_shared/src/callbacks.o
> In file included from ../src/callbacks.cc:18:0:
> ../src/openzwave.hpp:29:21: fatal error: Manager.h: Bestand of map bestaat niet
>  #include "Manager.h"
>                      ^
> compilation terminated.
> openzwave_shared.target.mk:115: recept voor doel 'Release/obj.target/openzwave_shared/src/callbacks.o' is mislukt
> make: *** [Release/obj.target/openzwave_shared/src/callbacks.o] Fout 1
> make: Map '/home/pi/.node-red/node_modules/openzwave-shared/build' wordt verlaten
> gyp ERR! build error
> gyp ERR! stack Error: `make` failed with exit code: 2
> gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:276:23)
> gyp ERR! stack     at emitTwo (events.js:106:13)
> gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
> gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:215:12)
> gyp ERR! System Linux 4.9.35-v7+
> gyp ERR! command "/usr/bin/nodejs" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
> gyp ERR! cwd /home/pi/.node-red/node_modules/openzwave-shared
> gyp ERR! node -v v6.11.2
> gyp ERR! node-gyp -v v3.4.0
> gyp ERR! not ok
> npm WARN node-red-project@0.0.1 No repository field.
> npm WARN node-red-project@0.0.1 No license field.
> npm ERR! Linux 4.9.35-v7+
> npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "install" "node-red-contrib-openzwave"
> npm ERR! node v6.11.2
> npm ERR! npm  v3.10.10
> npm ERR! code ELIFECYCLE
> 
> npm ERR! openzwave-shared@1.4.5 install: `node-gyp rebuild`
> npm ERR! Exit status 1
> npm ERR!
> npm ERR! Failed at the openzwave-shared@1.4.5 install script 'node-gyp rebuild'.
> npm ERR! Make sure you have the latest version of node.js and npm installed.
> npm ERR! If you do, this is most likely a problem with the openzwave-shared package,
> npm ERR! not with npm itself.
> npm ERR! Tell the author that this fails on your system:
> npm ERR!     node-gyp rebuild
> npm ERR! You can get information on how to open an issue for this project with:
> npm ERR!     npm bugs openzwave-shared
> npm ERR! Or if that isn't available, you can get their info via:
> npm ERR!     npm owner ls openzwave-shared
> npm ERR! There is likely additional logging output above.
> 
> npm ERR! Please include the following file with any support request:
> npm ERR!     /home/pi/.node-red/npm-debug.log

**Comment by robertsLando** [comment]
Check issue #39 and read Readme.md instruction. Before using this node you need to have open-zwave library and also openzwave shared installed.

**Comment by NickBue** [comment]
Thanks robertsLando,

I shuld have read better, but without your instructions I would not be any further.
The node is installed, that's 1 thing. I haven't read anything out yet, but that will come.

My homekit node for one or another reason is not working anymore since i've installed the zwave node. I will try to re-intall that one and hope there is not conflict.

Br

**Comment by NickBue** [comment]
Maybe 1:

I don't have the **zwave controller** available.
Any idea?

Thx
![image](https://user-images.githubusercontent.com/41262357/42804118-443d9c02-89a8-11e8-8d2d-b3337791f899.png)

**Comment by robertsLando** [comment]
That doesn't mean it isn't aviable, it has that icon because controller is not a node but it is a configuration node. You create a z-wave controller by creating a z-wave client in z-wave-in or/and z-wave out node

**Comment by NickBue** [comment]
Ok thanks,

Then I only have one issue with my Homekit node, but thats not for here.

It should be easy, but maybe I'm missing the complete point.
I have an Aeon Z-Wave stick and a Fibaro Motion Sensor.

I have 'included' the sensor to the stick. This is succesfull because in Domoticz I can see they are connected.
![image](https://user-images.githubusercontent.com/41262357/42805824-1edb71f0-89ad-11e8-99b3-6c72f0efa6b4.png)

Beside 'looking', I did nothing in domoticz.

Now I thought I could go to Node-red, just use the Z-wave in node and every time there was movement, it would just pop up in my debug node. But my debug node stays empty.

![image](https://user-images.githubusercontent.com/41262357/42805806-14bdc2fe-89ad-11e8-8b9c-aa20d0098cd9.png)

In the end, I just want temperature and a signal of movement in node-red. This z-wave is the only wireless device I will have in my house automation project. 
The signals received here will be transmitted via modbus to my PLC. This last part I have allready done for all my other devices (homekit, logitech, roomba, ring, alexa) On this way I have information of any kind of device in my PLC making my automation project quite complete. 

I have searched the web, but I can hardly find any info or exemples of this z-wave node, probably because it should be easy.
Unfortunately it's not working yet, so any suggestion is verry welcome.

br

thx

**Comment by NickBue** [comment]
Ohw,
now it suddenly works all the time, I just need to do something about the delay.
Thank for the help, I'll figure it out now.

Br

Nick

**Comment by robertsLando** [comment]
I don't know if you can do so much about the delay, if that is the only one sensor you have you can set a lower pool interval but it's not a "best practise".

**Comment by NickBue** [comment]
Ok, almost embarrassed, but now my node-red doesn't start anymore ...

I had the flow running like the screenshot here above. I disconnected the AEON stickto make some easy adaptions via Domoticz in the fibaro motion sensor (with that interface it's easy to change settings of the devices)

After that, I re-connected the stick into the PI. Because I've read you best do a restart after connecting it I rebooted the pi.
Normally my node-red starts automatically, but it was not available via the webserver. So I started it manually but now I get error messages.

It makes no difference if the stick is connected or not.

This are the screenshots:

**If I just enter NODE-RED**
![image](https://user-images.githubusercontent.com/41262357/42820463-84fa95c8-89d6-11e8-8519-9057d3ba9a54.png)


**If I enter NODE-RED START**
![image](https://user-images.githubusercontent.com/41262357/42820483-95a74f06-89d6-11e8-8c02-0dbca748ddec.png)


It must have something to do with the network key, but I have no idea what. It's not changed on the stick and is still the default key.

I have read the file /etc/openzwave/options.xml, but there is no info there.

My Linux knowledge is almost nothing, so it's very easy to get stuck in all this stuff. That's whats happening now....

**Comment by robertsLando** [comment]
Try this: 

```bash
node-red-stop
cd ~/.node-red 
sudo rm flows_raspberrypi_cred.json
node-red-start
```

This should remove the stored credentials file and automatically recreate one once node-red boots, after running node-red-start command you will have to insert the network key another time in the client configuration. 

For next time my suggestion is to shutdown raspberry before remving the z-stick  `sudo shutdown -h now` than disconnect and reconnect the power cable to start the raspberry

**Comment by NickBue** [comment]
So much thanks for your help...

Now I'm in a infinite loop, this stays comming up.....


17 Jul 14:20:51 - [info] node-red-contrib-openzwave: 1.2.4
17 Jul 14:20:51 - [info] openzwave-shared: 1.4.5
17 Jul 14:20:52 - [info] Dashboard version 2.4.3 started at /ui
17 Jul 14:20:53 - [warn] ------------------------------------------------------
17 Jul 14:20:53 - [warn] [node-red-node-serialport/serialport] Error: Module version mismatch. Expected 46, got 48. (line:6)
17 Jul 14:20:53 - [warn] [node-red-contrib-modbustcp/modbus tcp] SyntaxError: Block-scoped declarations (let, const, function, class) not yet supported outside strict mode
17 Jul 14:20:53 - [warn] [node-red-contrib-homekit/homekit] Error: Module version mismatch. Expected 46, got 48.
17 Jul 14:20:53 - [warn] ------------------------------------------------------
17 Jul 14:20:53 - [info] Settings file  : /home/pi/.node-red/settings.js
17 Jul 14:20:53 - [info] User directory : /home/pi/.node-red
17 Jul 14:20:53 - [warn] Projects disabled : set editorTheme.projects.enabled=true to enable
17 Jul 14:20:53 - [info] Flows file     : /home/pi/.node-red/flows_raspberrypi.json
17 Jul 14:20:53 - [info] Server now running at http://127.0.0.1:1880/
17 Jul 14:20:53 - [warn]
---------------------------------------------------------------------
Your flow credentials file is encrypted using a system-generated key.
If the system-generated key is lost for any reason, your credentials
file will not be recoverable, you will have to delete it and re-enter
your credentials.
You should set your own key using the 'credentialSecret' option in
your settings file. Node-RED will then re-encrypt your credentials
file using your chosen key the next time you deploy a change.
---------------------------------------------------------------------
17 Jul 14:20:53 - [info] Starting flows
Initialising OpenZWave 1.4.79 binary addon for Node.JS.
OpenZWave Security API is ENABLED
ZWave device db    : /usr/etc/openzwave
User settings path : /home/pi/.node-red
Option Overrides : --Logging false --ConsoleOutput false --QueueLogLevel 6 --DriverMaxAttempts 3 --NetworkKey 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F, 0x10
2018-07-17 14:20:53.870 Error, Cannot find a path to the configuration files at /usr/etc/openzwave/, Using /etc/openzwave/ instead...
2018-07-17 14:20:53.871 Info, Reading /etc/openzwave/options.xml for Options
2018-07-17 14:20:53.871 Warning, Failed to Parse /home/pi/.node-red/options.xml: Failed to open file
Error: Driver.cpp:6727 - Failed to Read Network Key
nodered.service: main process exited, code=exited, status=103/n/a
Unit nodered.service entered failed state.
nodered.service holdoff time over, scheduling restart.
Stopping Node-RED graphical event wiring tool....
Starting Node-RED graphical event wiring tool....
Started Node-RED graphical event wiring tool..
17 Jul 14:20:57 - [info]



![image](https://user-images.githubusercontent.com/41262357/42823741-989184a4-89de-11e8-82b7-7e37d706b6b5.png)

**Comment by robertsLando** [comment]
Try to also delete the flow file, so run: `sudo rm flows_raspberrypi*`

ATTENTION: This will delete all your flows in node, If the flow was easy and you remeber how to rebuild it do this than let me know

**Comment by NickBue** [comment]
Thanks man,

it's back running :)
after every adaptation I do a backup, luckily, because I have spend ages getting my flow all streamlined with my application.

I hope I can now smoothly go further with ZWave and not too many issues anymore.

Just because this all seems so easy for you, any idea whats going on with my homekit node? 
If not, I'll go searching via another topic.
![image](https://user-images.githubusercontent.com/41262357/42829334-d404ea3e-89e9-11e8-8a39-76366068ed9c.png)

Thanks again!

**Comment by NickBue** [comment]
Nevermind, I wass able to delete all my flows with:

    pi@raspberrypi:~/.node-red $ rm flows_raspberrypi.json


Any idea  though? Is it in my z-WAVE stick that I have an encription or is it really in the program?
Just to know if i should reset te stick or restart my program code

**Comment by NickBue** [comment]
Hi,

Not sure what you mean with 'remove the secret network key'.

- I imported my backup
- Deleted the pink zwave node in my flow AND the configuration node which was also back there
- After that, I actually had nothing of zwave in my project.
- I deployed without any issue
- I've replaced the zwave stick and just to be sure, again I could deploy the project without zwave
- I inserted a NEW zwave node, and I only configured the port. Now it crashes after a deploy. I never have placed a network key here.

![image](https://user-images.githubusercontent.com/41262357/47380201-2e7fd680-d6fd-11e8-9017-b32276d0e91d.png)

Next step is simply to reset the stick, but then I need to reconfigure and also reset every zwave device. So if there is one more thing to try?

br
